src/branch_and_bound.rs

```diff
@@ -0,0 +1,582 @@
+use bitcoin::Amount;
+
+use crate::WeightedUtxo;
+use bitcoin::amount::CheckedSum;
+use bitcoin::blockdata::effective_value;
+use bitcoin::FeeRate;
+
+fn index_to_utxo_list(
+    index_list: Option<Vec<usize>>,
```

Comment:
You could accept `Option<&[usize]>` here to allow reusing the allocation in the caller.

Reply:
I'm not sure this will change performance since since this function isn't really in the "hot path" and only gets called once when the function has finished, unless I'm missing something.  Although it can't hurt either to do this.

Reply:
Oh, then it may be best to keep it as is.